docs/design/import.md

```diff
@@ -0,0 +1,314 @@
+// it won't be importable.
+//
+// "Importing" in Terraform is the process of taking an already-created
+// resource and bringing it under Terraform management. This can include
+// updating Terraform state, generating Terraform configuration, etc.
+type ResourceImporter struct {
+    // State is called to convert an ID to one or more InstanceState to
+    // insert into the Terraform state.
+    //
+    // Deprecated: State is deprecated in favor of StateContext.
+    // Only one of the two functions can bet set.
+    State StateFunc
+
+    // StateContext is called to convert an ID to one or more InstanceState to
+    // insert into the Terraform state. If this isn't specified, then
+    // the ID is passed straight through. This function receives a context
+    // that will cancel if Terraform sends a cancellation signal.
+    StateContext StateContextFunc
+}
+
+// StateFunc is the function called to import a resource into the Terraform state.
+//
+// Deprecated: Please use the context aware equivalent StateContextFunc.
+type StateFunc func(*ResourceData, interface{}) ([]*ResourceData, error)
+
+// StateContextFunc is the function called to import a resource into the
+// Terraform state. It is given a ResourceData with only ID set. This
+// ID is going to be an arbitrary value given by the user and may not map
+// directly to the ID format that the resource expects, so that should
+// be validated.
+//
+// This should return a slice of ResourceData that turn into the state
+// that was imported. This might be as simple as returning only the argument
+// that was given to the function. In other cases (such as AWS security groups),
+// an import may fan out to multiple resources and this will have to return
+// multiple.
+//
+// To create the ResourceData structures for other resource types (if
+// you have to), instantiate your resource and call the Data function.
+type StateContextFunc func(context.Context, *ResourceData, interface{}) ([]*ResourceData, error)
+```
+
+When defined, the resource could respond to the `ImportResource` RPC, otherwise an error was returned that the resource does not support import.
+
+If the resource state could be entirely fetched using an import identifier that matched the resource identifier, the `ImportStatePassthrough` and `ImportStatePassthroughContext` helpers simplified provider implementations, e.g.
+
+```go
+Importer: &schema.ResourceImporter{
+    State: schema.ImportStatePassthrough,
+},
+```
+
+Otherwise, custom provider implementations were required.
+
+## Caveats
+
+Terraform supports the ability to import multiple resource states during a single import (for example, resource import of an `aws_s3_bucket` also imports any associated `aws_s3_bucket_policy`), however almost all resources in the Terraform ecosystem with import support implement a 1:1 import. The extremely uncommon multiple resource support in practice is likely due to:
```

Comment:
This last sentence

> The extremely uncommon multiple resource support in practice is likely due to:

sounded to me like a list of reasons why people would want to do that was going to follow, but it looks like a list of why that's a bad idea to do? Or why it's hard to use? I'm not clear.